frontend/src/app/side-menu/side-menu.component.ts

```diff
@@ -77,9 +84,29 @@ export class SideMenuComponent {
     formData.append('file', file.data);
     file.inProgress = true;
 
+    this.currUpload = true;
     this.sharedService
       .sendFormData(formData, '/upload')
       .subscribe((event: any) => {
+        // Display progress bars to the user.
+        switch (event.type) {
+          case 1:
```

Comment:
What are 1 and 3? You should have these variables defined somewhere.

Reply:
Also, should the switch have a default condition?

Reply:
Type 1 events represent upload progress and type 3 events represent download progress. Since those are the only two events relevant for the progress bar tracking I don't think we need a default in the switch

Reply:
Can't you use HttpEventType::UploadProgress instead of 1 like you'd in C++?

An empty default i usually still advisable.

Reply:
Good idea,  I can do that and implemented it.